tests/ui/offset-of/offset-of-dst-field.rs

```diff
@@ -0,0 +1,14 @@
+#![feature(offset_of)]
+
+use std::mem::offset_of;
+
+#[repr(C)]
+struct Foo {
+    x: u8,
+    y: u16,
+    slice: [u8],
+}
+
+fn main() {
+    offset_of!(Foo, slice); //~ ERROR the size for values of type
+}
```

Comment:
I don't think this should error. Slice has a statically known alignment and offset, this should be allowed. You only need `Sized` for `span_of`, for `offset_of` statically known alignment is enough.

(note that `dyn Trait` does [**not**](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=66ddae25b86bba5d4d4818e66f993243) have a statically known alignment, but for slices and [extern types](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=b621fc501149c9689334bd8fb21d739c) do)

---

<sup>this also makes me wander if we should have `offset_of_val` or something, to support `dyn Trait` & co (not in this PR obv, but maybe eventually via a new RFC...)</sup>

Reply:
That being said, this is listed as "future possibility" in the rfc https://rust-lang.github.io/rfcs/3308-offset_of.html#-by-only-forbidding-the-edge-case (option `2` here), so a `// FIXME` somewhere (ideally on the `Sized` check) mentioning that we could extend this is also fine.